pos_session_close_by_tax/models/pos_order.py

```diff
@@ -0,0 +1,90 @@
+from odoo import models, _
+from odoo.tools import float_round
+
+
+class Order(models.Model):
+    _inherit = 'pos.order'
+
+    def _create_account_move_line(self, session=None, move=None):
+        res = super(Order, self)._create_account_move_line(session, move)
+        if move:
+            self._group_lines_by_tax(move)
+        return res
+
+    @staticmethod
+    def _check_grouping_applicability(lines):
```

Comment:
```suggestion
    @api.model
    def _check_grouping_applicability(self, lines):
```

Same for others.
Besides it being "the standard" in odoo code, it makes it easier for other possible modules extending this one to do things like

```py
    def _check_grouping_applicability(self, lines):
        if not self.env.context.get("skip_check"):
            return super()._check_grouping_applicability(lines)
```

Suggestion also applied to the other methods with `@staticmethod`